streams/authorizer/src/main/java/com/hortonworks/streamline/streams/security/authentication/StreamlineKerberosRequestFilter.java

```diff
@@ -46,7 +47,10 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
                 httpRequest.getMethod(), httpRequest.getAuthType(),
                 httpRequest.getRemoteUser(), principal, scheme);
 
-        if (principal == null || !httpRequest.getAuthType().equalsIgnoreCase(KERBEROS_AUTH)) {
+        // We now also support Knox SSO using jwt as underlying authentication mechanism. Hence added jwt as a valid type
+        // Probably jwt does not sound right in this class and we are also passing KERBEROS_AUTH for jwt below where we
+        // instantiate a SecurityContext. That is because there are checks down the line for kerberos as well
```

Comment:
Maybe then this class needs to be renamed to `StreamlineAuthenticationRequestFilter` and the right `authType`  passed to the StreamlineSecurityContext as well?